README.md

```diff
@@ -0,0 +1,16 @@
+# bird-monitor
+
+bird-monitor is an android app that makes periodic sound recordings and uploads them to the cloud, noting the time and location.
+These recordings will be analysed to detect the volume of birdsong and how it changes over time, giving you an objective measure of
+ the impact of predator control or reforestation.
+
+## License
+
+This project is licensed under the Affero General Public License
+(https://www.gnu.org/licenses/agpl-3.0.en.html).
+
+## Releases
+
+* Ensure all required changes have been merged into the master branch on Github.
+* Ensure you have updated app/build.gradle version parameters eg. (ext.versionPatch)
+* VersionCode and VersionName will be automatically generated
```

Comment:
Would probably be good to elaborate a bit more here on what triggers a release, as it's a bit unclear to me. I think we still need to run `fastlane deploy` right? How do we trigger that, is it on tags from master branch? Or does it just be triggered locally by someone with the JSON secret key in that dir? Who has access to that key?

Also a bit about the beta release, as that's unclear to me too what that does and when it is/should be triggered.

Thanks :)

Reply:
What triggers a release: Up to now, I've been doing a release when convenient/some feature(s) have been added that have been asked for.  It has also been dependent on how confident I was that no bugs were in the code - especially none that affected the core ability to do hourly and dawn dusk recordings.  I will normally/ideally want to get feedback from others that the functionality was as expected and then put a phone on extended test - say for a week.

I am totally open to being advised on a different/more 'normal' approach - as long as sufficient testing is still part of the release process.

Signing/keys: Menno/Cameron/me have the keystore/password for signing - up to Menno on this.

Release names: Recently I've learned more about how to do this from https://en.wikipedia.org/wiki/Software_versioning:
  
My thoughts are that we have major.minor.maintenance . Minor increases when new features are added, and maintenance increases with bug fixes.  Not sure what a major increment will look like - but probably a significant change to the look?

I'm thinking that when we want others outside of the Bird-Monitor developers to test a release we first make it a pre-release, which then gets changed to an actual release if it passes all feedback/testing.  If it does not pass, then it is just left as a pre-release.

What branch to create release versions from? I'm thinking that we create a new branch for when we want to create a release - I think I read somewhere that that was a good way - but happy to be told otherwise.

I'm new to all this, so most happy to be told how to so all this :-)

Reply:
I think we would use fastlane deploy I wanted to find out how we were releasing at the moment first. before I write new instructions. Will catch up with Menno and Cam tomorrow to discuss.

I believe in Fastlane/Fastfile we can define custom deployments e.g. Pre-Releases.

Reply:
Also to do releases will need access to the json key file for the play account.